modules/environment/shell.nix

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2019-2021, see AUTHORS. Licensed under MIT License, see LICENSE.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+
+  ###### interface
+
+  options = {
+
+  };
+
+
+  ###### implementation
+
+  config = {
+
+    environment.etc = {
+      "profile".text = ''
+        . "${config.build.sessionInit}"
```

Comment:
It's a directory containing `etc/profile.d/nix-on-droid-session-init.sh`.

Reply:
Oh right. Then why did it work for me.. I will have a look at that..

Reply:
Fixed the description and type of sessionInit option :)

Reply:
Thank you.